sztp-agent/pkg/secureAgent/utils.go

```diff
@@ -20,7 +29,79 @@ func linesInFileContains(file string, substr string) string {
 	return ""
 }
 
-func extractURLfromLine(line, regex string) string {
+func extractfromLine(line, regex string, index int) string {
 	re := regexp.MustCompile(regex)
-	return re.FindAllString(line, -1)[1]
+	return re.FindAllString(line, -1)[index]
+}
+
+func (a *Agent) doTLSRequestToBootstrap() (*BootstrapServerPostOutput, error) {
+
+	body := strings.NewReader(a.GetInputJSONContent())
+	log.Println(a.GetInputJSONContent())
+	r, err := http.NewRequest(http.MethodPost, a.GetBootstrapURL(), body)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Println(a.GetSerialNumber(), a.GetDevicePassword())
+	r.SetBasicAuth(a.GetSerialNumber(), a.GetDevicePassword())
+	r.Header.Add("Content-Type", a.GetContentTypeReq())
+
+	caCert, _ := ioutil.ReadFile(a.GetBootstrapTrustAnchorCert())
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+	cert, _ := tls.LoadX509KeyPair(a.GetDeviceEndEntityCert(), a.GetDevicePrivateKey())
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:      caCertPool,
+				Certificates: []tls.Certificate{cert},
+			},
+		},
+	}
+	log.Println(r)
+	time.Sleep(20 * time.Second)
+	res, err := client.Do(r)
+	defer res.Body.Close()
+
+	post := &BootstrapServerPostOutput{}
+	derr := json.NewDecoder(res.Body).Decode(post)
```

Comment:
I also used for debug to print the Body this code:
```
        b, err := io.ReadAll(res.Body)
        log.Println(string(b))
```

Reply:
I've got the response struct log, the same but in struct